### ISSUE TITLE
Add Double Pareto-Lognormal Distribution

### ISSUE THREAD

**Issue by cancan101** [scipy/scipy#10364]
Add the [Double Pareto-Lognormal Distribution](https://www.math.uvic.ca/faculty/reed/dPlN.3.pdf) (Reed, 2003) to scipy which is described as: 
>A family of probability densities, which has proved useful in modelling the size distributions of various phenomens, including incomes and earnings, human settlement sizes, oil-field volumes and particle sizes, is introduced.

I found this implementation. (_Link removed due to GPL license incompatibility._)

There are [a number of well cited papers](https://scholar.google.com/scholar?q=double+pareto-lognormal+distribution&hl=en&as_sdt=0&as_vis=1&oi=scholart) that use this distributions to model various empirical distributions.

**Comment by chrisb83** [comment]
The distribution qualifies for inclusion into SciPy. The first paper also gives a way to generate rvs. Are you interested to submit a PR? Or is it a Feature requests?

**Comment by cancan101** [comment]
This paper has some good notes on fitting https://core.ac.uk/download/pdf/55654372.pdf

**Comment by genccal** [comment]
what's the follow-up? Someone is gonna implement it?

**Comment by rlucas7** [comment]
Pull request welcome. 

-L

**Comment by dschmitz89** [comment]
I played around with this a little and think I found a way to implement sampling and PDF:

<details><summary>Details</summary>
<p>

```python
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
from scipy.special import log_ndtr as norm_logcdf

def norm_logsf(x):
    return norm_logcdf(-x)

def rvs(loc=0, scale=1, a=1, b=1, size=1):
    rvs = np.random.lognormal(mean=loc, sigma=np.sqrt(scale), size=size)
    scratch_space = stats.pareto(b=a).rvs(size=size)
    rvs *= scratch_space
    scratch_space[:] = stats.pareto(b=b).rvs(size=size)
    rvs /= scratch_space
    return rvs

def pdf(x, loc=0, scale=1, a=1, b=1):
    a_plus_b = a + b
    log_f1 = (np.log(a) + np.log(x) * (-1. - a) + (a * loc + a**2 * scale / 2.) +
                norm_logcdf((np.log(x) - loc - a * scale)/np.sqrt(scale)))
    log_f2 = (np.log(b) + np.log(x) * (b -1.) + (-b * loc + b**2 * scale / 2.) +
              norm_logsf((np.log(x) - loc + b * scale)/np.sqrt(scale)))

    return b/a_plus_b * np.exp(log_f1) + a/a_plus_b * np.exp(log_f2)

loc=3
scale=1
a=1.5
b=2.
n = 1000000
n_bins = int(np.sqrt(n))
samples = rvs(loc, scale, a, b, size=n)
xmin, xmax = np.percentile(samples, 0.01), np.percentile(samples, 99.)
bins = np.logspace(np.log(xmin), np.log(xmax), n_bins)
plt.hist(samples, density=True, range=(xmin, xmax), bins=bins)#)
plt.plot(bins, pdf(bins, loc, scale, a, b), ls="dashed")
print(samples.max(), samples.min())
plt.xscale('log')
plt.show()
``` 

</p>
</details> 

![image](https://github.com/scipy/scipy/assets/40656107/1043503e-6c52-4bcc-b40c-3eca27fbe6c4)

I have to see how nicely this plays with the loc scale parametrization from scipy. Would one of the usual suspects have time to review in the coming weeks? CC @tupui @chrisb83 @WarrenWeckesser

**Comment by mdhaber** [comment]
> I tried to implement this with SciPy's loc-scale distribution but cannot get the PDF/CDF formulas to work

More specifically, what couldn't you get to work? I implemented this independently and reproduced the plot you have. Were you concerned about apparent disagreement between the PDF and histogram in the left tail, or was it something else? (That just seems to be a numerical issue in the sampling, and I suspect that we wouldn't see that if we worked with the logarithm - the normal-Laplace - directly.)
